### ISSUE TITLE
Support for Randomization Inference

### ISSUE THREAD

**Issue by s3alfisc** [s3alfisc/pyfixest#427]
Introduce an `ritest()` method for `Feols` using similar syntax as @grantmcdermott's [ritest](https://github.com/grantmcdermott/ritest). 

```r
ritest(co_est, 'b_treat', strata='b_pair', cluster='b_block', reps=1e3, seed=1234)
#> 
#>           Call: feols(fml = dayscorab ~ b_treat + b_dayscorab + miss_b_dayscorab | b_pair + round2 + round3, data = colombia, vcov = ~b_block)
#>    Res. var(s): b_treat
#>             H0: b_treat=0
#>  Strata var(s): b_pair
#>         Strata: 31
#> Cluster var(s): b_block
#>       Clusters: 63
#>      Num. reps: 1000
#> ──────────────────────────────────────────────────────────────────────────────── 
#>   T(obs)         c         n     p=c/n     SE(p)   CI 2.5%  CI 97.5%  
#>  -0.1807       107      1000     0.107   0.01609   0.08054    0.1335  
#> ──────────────────────────────────────────────────────────────────────────────── 
#> Note: Confidence interval is with respect to p=c/n. 
#> Note: c = #{|T| >= |T(obs)|}
```

**Comment by s3alfisc** [comment]
Additionally, see the [paper ](https://hesss.org/ritest.pdf)and [Stata code ](https://github.com/simonheb/ritest) by @simonheb for details. 

Note: RI can also be used as input to the stepwise procedure for multiple-hypothesis correction by Romano and Wolf, as implemented in the [rwolf](https://github.com/s3alfisc/pyfixest/blob/4e29dafbc2f79daf625b490746f75856f8958ed7/pyfixest/estimation/multcomp.py#L66) function.

**Comment by s3alfisc** [comment]
Implemented in #431: 
- ritest for OLS, Poisson, one and two-sided testing via the randomization-c

To be added: support for 
- randomization-t
- WLS
- IV
- CIs via inversion
